services/graphql/src/mongoose/schema/ad.js

```diff
@@ -225,11 +225,9 @@ schema.pre('save', async function setReady() {
 schema.pre('save', async function updateEvents() {
   if (this.isModified('lineitemId')) {
     const { advertiserId, orderId, lineitemId } = this;
-    ['view', 'click'].forEach((modelName) => {
-      connection.model(modelName).updateMany({ adId: this._id }, {
-        $set: { advertiserId, orderId, lineitemId },
-      }).catch(e => logError(e));
-    });
+    connection.model('event').updateMany({ adId: this._id }, {
```

Comment:
This query must also include the proper types, e.g:
`{ adId: this._id, type: { $in: ['view', 'click']  } }`

Reply:
Index this as well